packages/fs-detectors/src/detect-framework.ts

```diff
@@ -135,6 +135,31 @@ async function matches(
   };
 }
 
+function removeSupercededFramework(
+  matches: (Pick<Framework, 'supercedes' | 'slug'> | null)[],
+  slug: string
+) {
+  const index = matches.findIndex(f => f?.slug === slug);
+  if (index !== -1) {
+    const framework = matches[index]!;
+    if (framework.supercedes) {
+      removeSupercededFramework(matches, framework.supercedes);
+    }
+    matches.splice(index, 1);
+  }
+}
+
+function removeSupercededFrameworks(
+  matches: (Pick<Framework, 'supercedes' | 'slug'> | null)[]
+) {
+  for (const match of matches.slice()) {
+    if (!match) continue;
+    if (match.supercedes) {
+      removeSupercededFramework(matches, match.supercedes);
+    }
+  }
+}
```

Comment:
<!-- Suggestion comments propose improvements to the code. Mention specifically what should be changed and what the benefit would be.  -->
**suggestion (soft-blocking):** add unit tests

In this or a follow-up PR: Can we add some unit tests around this behavior?‏

It took me some careful reading to understand exactly what was happening here. This logic also supports chained superseding, which isn't currently tested.